hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1279,13 +1279,9 @@ public function upgrade_1030() {
     try {
       CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` ADD PRIMARY KEY (`id`)');
       CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` CHANGE `id` `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT');
-    } catch (Exception $e) {
-      $tableStructure = CRM_Core_DAO::executeQuery('DESC civicrm_hrhours_location');
-
-      while ($tableStructure->fetch()) {
-        if ($tableStructure->Field == 'id' && ($tableStructure->Key != 'PRI' || stripos($tableStructure->Extra, 'auto_increment') === false)) {
-          throw $e;
-        }
+    } catch (PEAR_Exception $e) {
+      if (stripos($e->getCause()->userinfo, 'nativecode=1068') === false) {
```

Comment:
Yeah, the original message for the exception is very obscure, it just says 'Unknown Error'. $e->getCause()->userinfo holds actual details of error, which are a lot more helpful if an error occurs.